variable/comparison.cpp

```diff
@@ -4,16 +4,35 @@
 /// @author Piotr Rozyczko
 #include "scipp/core/element/comparison.h"
 #include "scipp/variable/comparison.h"
+#include "scipp/variable/math.h"
 #include "scipp/variable/transform.h"
+#include "scipp/variable/util.h"
 #include "scipp/variable/variable.h"
 
 using namespace scipp::core;
 
 namespace scipp::variable {
 
-Variable is_approx(const VariableConstView &a, const VariableConstView &b,
-                   const VariableConstView &t) {
-  return transform(a, b, t, element::is_approx);
+Variable is_close(const VariableConstView &a, const VariableConstView &b,
+                  const VariableConstView rtol, const VariableConstView atol,
+                  const NanComparisons equal_nans) {
+  const auto tol = atol + rtol * abs(b);
+  if (a.hasVariances() && b.hasVariances()) {
+    const auto error_tol = atol + rtol * abs(variances(b));
+    if (equal_nans == NanComparisons::Equal)
+      return variable::transform(a, b, tol, element::is_close_equal_nan) &
+             variable::transform(sqrt(variances(a)), sqrt(variances(b)),
+                                 error_tol, element::is_close_equal_nan);
+    else
+      return variable::transform(a, b, tol, element::is_close) &
+             variable::transform(sqrt(variances(a)), sqrt(variances(b)),
+                                 error_tol, element::is_close);
```

Comment:
```suggestion
  if (a.hasVariances() && b.hasVariances()) {
    return is_close(values(a), values(b), rtol, atol, equal_nans) & is_close(sqrt(variances(a)), sqrt(variances(b)), rtol, atol, equal_nans);
```
Much shorter and less error prone? I also think `error_tol` was wrong?

Reply:
But with this suggestion, it uses the same tolerance for values and variances.

Reply:
Yes, if you mean stddev (not variances), isn't that what it should do? Otherwise we need to think about factors of 2 etc?

Reply:
I think @jl-wynen is referring to the `rtol * abs(variances(b))` product?

Reply:
Yes, me too.

Reply:
Wasn't that the idea of applying rtol to values and variances separately as in Owen's implementation?

Reply:
I don't understand, Owen's original implementation was based on the same `rtol` for the variances, not based on an extra argument?

Reply:
Yes, the relative tolerance is the same. But it is applied to `values(y)` for comparing values and to `variances(y)` for comparing variances.

Now that I looked at your suggestion again, which `is_close` is this calling?

Reply:
Not the same: Mine has `sqrt`, so the behavior of `rtol` is different from your implementation.

Reply:
Having the same atol is a bit dubious, but I don't propose we try to address this as users can zero that out.